### PR TITLE
Mavenizing async-profiler Java library

### PR DESCRIPTION
**Pull request by JigarJoshi**
#93 

Step 1: Mavenize the Java library <-- this PR takes care for this
Step 2: Create friendly Java API (take care of extracting and loading library .so)
Step 3: publish it to maven central

### CODE REVIEW

**Comment by incubos** [comment]
Could you please temporarily publish the artifact from [your fork using JitPack](https://jitpack.io/#JigarJoshi/async-profiler) and test it (i.e. the library is loaded successfully)?

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        <phase>package</phase>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                            <executable>/usr/bin/zip</executable>
+                            <arguments>
+                                <argument>-u</argument>
+                                <argument>target/async-profiler-${project.version}.jar</argument>
+                                <argument>build/libasyncProfiler.so</argument>
```
Comment (incubos) [inline pom.xml:62]: `async-profiler` supports [many platforms](https://github.com/jvm-profiling-tools/async-profiler#download) having prebuilt libraries and Maven artifact we are going to publish is expected to be used from any supported platform, so we have to find a way to pack all the binaries.

**Review by incubos — CHANGES_REQUESTED** [review]
Multiplatform support is needed if we are going to publish a single artifact.

**Comment by apangin** [comment]
Hi,
Sorry for the delay - that was a rather busy month for me.

First, please create PRs against v2.0 branch - it will become mainline soon. No new features are planned for 1.8 - it will remain only for bugfixes.

My main concern is how to package async-profiler agent binaries. As Vadim noticed, async-profiler supports many platforms, each requiring a separate binary. In future, the number of platforms will likely increase (think of Apple M1, for example). I don't like an idea of packaging all possible binaries into a single JAR, otherwise it'll get enormously large. Furthermore, I expect that some users will prefer to install async-profiler package with the OS-specific package manager like RPM, and some will build the profiler on their own. Therefore, I think it makes sense to create a separate Maven artifact with the API only, and separate artifacts with `.so` for each individual platform.

Side note: by default, `make` builds the profiler agent, `jattach` helper, API jar, and converters jar. When preparing `libasyncProfiler.so`, we may build just the agent library and skip other steps.

**Comment by apangin** [comment]
Closing PR, since it has been superseded by the ongoing work within #93.

**Pull request closed without merging**